internal/workflow/processing.go

```diff
@@ -205,20 +205,20 @@ func (w *ProcessingWorkflow) Execute(ctx temporalsdk_workflow.Context, req *inge
 		var sessErr error
 		maxAttempts := 5
 
-		activityOpts := temporalsdk_workflow.WithActivityOptions(ctx, temporalsdk_workflow.ActivityOptions{
+		ctx = temporalsdk_workflow.WithActivityOptions(ctx, temporalsdk_workflow.ActivityOptions{
```

Comment:
Adding them to the main context allows the recreate session to work without setting this activity options again.